discovery-handlers/opcua/src/discovery_impl.rs

```diff
@@ -180,6 +192,38 @@ fn get_socket_addr(url: &str) -> Result<SocketAddr, anyhow::Error> {
     Ok(addr)
 }
 
+// This checks if the discovery_url can be resolved, if not use ip address instead
+fn get_discovery_url_ip(ip_url: &str, discovery_url: String) -> Result<String, anyhow::Error> {
+    let url = Url::parse(&discovery_url).map_err(|_| anyhow::format_err!("could not parse url"))?;
```

Comment:
Might be helpful to add the url to the error
```suggestion
    let url = Url::parse(&discovery_url).map_err(|_| anyhow::format_err!("could not parse url {discovery_url}"))?;
```

Reply:
thanks! updated to log discovery_url in the error msg